[VBLOCKS-1530] Feat: readme 3 rd party auth (#35)
twilio/twilio-voice-react-native-app: README.md

Author: Yea it did. Are you and @charliesantos using different linters?

Reviewer: I have whatever is set up on the project.

Author: Seems like we have a mix of `*` and `-` https://github.com/twilio/twilio-voice-react-native-app/blob/main/README.md?plain=1 . I also did not customize anything on my end.

Reviewer: I'm ok with either. As long as we're consistent.

## Features

- Dialpad for making outgoing calls to a phone number
- Active call screen to interact with and control ongoing calls
- Login flow for authenticating and authorizing users

### Planned Features

- App to app calls
- Registration and notifications for incoming calls

![dialer](https://user-images.githubusercontent.com/22135968/223575482-53f733b4-c828-4dd1-b100-7cc945a52c9f.jpg)

> ![active](https://user-images.githubusercontent.com/22135968/223575472-f112ab20-f11c-4f54-9432-3640ddbb0f1c.jpg)

Reviewer: Let's add a screenshot of the login screen, right before the dialpad screen

* Dialpad for making outgoing calls to a phone number
* Active call screen to interact with and control ongoing calls
* Login flow for authenticating and authorizing users

### Planned Features

* App to app calls
* Registration and notifications for incoming calls

![dialer](https://user-images.githubusercontent.com/22135968/223575482-53f733b4-c828-4dd1-b100-7cc945a52c9f.jpg)
![active](https://user-images.githubusercontent.com/22135968/223575472-f112ab20-f11c-4f54-9432-3640ddbb0f1c.jpg)

> ![login](https://user-images.githubusercontent.com/35968892/226741082-eb0ac83a-b16d-46a3-9ff3-a83167ed7267.png)

Reviewer: Let's put this right before the dialpad screen. Also, can you please take a screenshot on iOS to be more consistent? Feel free to send the screenshot to me and I can clean up the edges as well.

## Prerequisites

* Understanding of [Twilio Programmable Voice](https://www.twilio.com/docs/voice/sdks).
* A [React Native Development Environment](https://reactnative.dev/docs/0.70/environment-setup).

> * Latest LTS version of [NodeJS](https://nodejs.org/en/).

Reviewer 2: Not part of your changes, but I think the official RN docs actually recommend LTS 16 at least for the version of RN we use. There's a `.node-version` file in the project root that is put there by the RN boilerplate, and I've personally had issues with versions of Node that aren't the major version specified in that file. Let's note this as a future task? @charliesantos @kpchoy what do you think?

* A Node package manager. Examples in this project will use the latest version of [Yarn v1]https://classic.yarnpkg.com/lang/en/.
* Create an [Auth0 Account](https://auth0.com/signup?place=header&type=button&text=sign%20up)

## Setting Up Auth0 Authentication

> The Twilio Voice React Native Reference App implements Auth0 to demonstrate login flow for authenticating and authorizing users. Sign In to your Auth0 account and follow the steps below.

Reviewer 2: ```suggestion
The Twilio Voice React Native Reference App implements Auth0 to demonstrate login flow for authenticating and authorizing users. Sign in to your Auth0 account and follow the steps below.
```
Minor nit, personally prefer.

### Auth0 for the app

1. Create a new Auth0 `Native` app from the Auth0 user console, then enter your desired `Name`

![auth0reactnative](https://user-images.githubusercontent.com/35968892/226486988-5dc0172d-0f46-4957-916b-385c2bdeaf8b.png)

> 2. Navigate to the settings tab of your newly created Auth0 Application. You will now need to fill out the fields labeled `Allowed Callback Urls` and `Allowed Logout Urls`. Depending on your use case enter the ios format, android format, or both.

Reviewer 2: ```suggestion
2. Navigate to the settings tab of your newly created Auth0 Application. You will now need to fill out the fields labeled `Allowed Callback Urls` and `Allowed Logout Urls`. Depending on your use case enter the iOS format, Android format, or both.
```

- A [React Native Development Environment](https://reactnative.dev/docs/0.70/environment-setup).
- Latest LTS version of [NodeJS](https://nodejs.org/en/).
- A Node package manager. Examples in this project will use the latest version of [Yarn v1](https://classic.yarnpkg.com/lang/en/).

> - Create a free tier [Auth0 Account](https://auth0.com/signup?place=header&type=button&text=sign%20up)

Reviewer: I don't think free tier is a requirement. We can just say `An Auth0 account`

## Setting Up Auth0 Authentication

> This app implements Auth0 to demonstrate login flow for authenticating and authorizing users
> 
> Sign In to your Auth0 account

Reviewer: ```suggestion
Twilio Voice React Native Reference implements Auth0 to demonstrate login flow for authenticating and authorizing users. Sign In to your Auth0 account and follow the steps below.
```

Reviewer 2: ```
The Twilio Voice React Native Reference...
```
on top of what Charlie recommends.

> ### Auth0 for the client

Reviewer: ```suggestion
### Auth0 for the app
```

> Create a new Auth0 `Native` app from the Auth0 user console
> 
> - Input your desired `Name`

Reviewer: There's only one step. We can probably combine this. Also, prefer `enter` over `input`.
```suggestion
Create a new Auth0 `Native` app from the Auth0 user console and enter your desired `Name`
```

![auth0reactnative](https://user-images.githubusercontent.com/35968892/226486988-5dc0172d-0f46-4957-916b-385c2bdeaf8b.png)

### Auth0 for the server

> Create a new Auth0 `API` from the Auth0 user console
> 
> - Input your desired `Name`
> - Input your desired `Identifier`, recommend use pattern `https://{Name}/api`

Reviewer: ```suggestion
Create a new Auth0 `API` from the Auth0 user console. Then, enter your desired `Name` and `Identifier`.
```
